dashboard/modules/reporter/reporter_head.py

```diff
@@ -411,6 +411,103 @@ async def get_task_cpu_profile(self, req) -> aiohttp.web.Response:
             headers={"Content-Type": "text/html"},
         )
 
+    @routes.get("/task/memory_profile")
+    async def task_memory_profile(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the memory profile for a specific task.
+        Note that one worker process works on one task at a time
+        or one worker works on multiple async tasks.
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the memory profile data.
+
+        Raises:
+            ValueError: If the "task_id" parameter is
+            missing in the request query.
+            ValueError: If the "attempt_number" parameter is
+            missing in the request query.
+            ValueError: If the maximum duration allowed is exceeded.
+            ValueError: If the worker begins working on
+            another task during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is
+            an internal server error during the profile retrieval.
+        """
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+        if "node_id" not in req.query:
+            raise ValueError("node_id is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+        node_id = req.query.get("node_id")
+
+        duration = int(req.query.get("duration", 30))
```

Comment:
Could we rename `duration` to `duration_s` throughout the code, so it's clear that `duration` is in seconds?